openeogeotrellis/job_tracker_v2.py

```diff
@@ -85,48 +87,112 @@ class YarnAppReportParseException(Exception):
 class YarnStatusGetter(JobMetadataGetterInterface):
     """YARN app status getter"""
 
+    def get_application_url(self, application_id: str) -> str:
+        return url_join(
+            ConfigParams().yarn_rest_api_base_url,
+            f"/ws/v1/cluster/apps/{application_id}",
+        )
+
+    def get_authentication_provider(self):
+        auth = None
+        if ConfigParams().yarn_auth_use_kerberos:
+            # TODO: HTTPKerberosAuth is deprecated, this is a utility class for compatibility with requests_kerberos.
+            #   Find the better way to do this, using requests_gssapi directly.
+            # Have to set mutual_authentication to REQUIRED or it just skips kerberos auth it seems.
+            auth = requests_gssapi.HTTPKerberosAuth(
+                mutual_authentication=requests_gssapi.REQUIRED
+            )
+        return auth
+
     def get_job_metadata(self, job_id: str, user_id: str, app_id: str) -> _JobMetadata:
-        try:
-            command = ["yarn", "application", "-status", app_id]
-            with TimingLogger(f"Running {command}", logger=_log.debug):
-                application_report = subprocess.check_output(
-                    command, stderr=subprocess.PIPE
-                ).decode("utf-8")
-        except CalledProcessError as e:
-            stdout = e.stdout.decode()
-            if "doesn't exist in RM or Timeline Server" in stdout:
-                raise AppNotFound(stdout) from None
-            else:
-                raise
-        return self.parse_application_report(application_report)
+        # TODO: Are there any cases were we *do* need to provide credentials to the Kerberos authentication?
+        #   At present we are assuming there is already a Kerberos ticket because kinit was run before
+        #   the geopyspark driver process was started.
+        #   AFAIK that means we don't have to provide user & pw which is in fact the whole point.
```

Comment:
Indeed, the `kinit` takes care of this so this TODO can be removed.

Reply:
Done.